### PR TITLE
On This Day Widget

### PR DESCRIPTION
**Pull request by mcleinman**
**Fixes Phabricator ticket:** 

### Notes
* 

### Test Steps
1. Test English. Test a non-English LTR language (like French). Test a RTL language (like Arabic). 
1. Ensure when you tap on the widget, it goes to appropriate page. And if you tap the article box on the large widget, it takes you there.
1. Try a language that doesn't support On this day. (Most of them do not support it.) It should give a nice error message.
1. Make text size smaller and larger, makes sure it's okay.
1. No connection: Since it will normally use the most recent valid hit, to test the "no internet" error message you should uninstall the app, turn on airplane mode, install the app, and add a widget. (And if this is the only possible way to see this error message, it's overkill. It's in there in case it ever tries to refresh when there is no internet, and it doesn't fall back on the last valid data.)
1. Placeholder text: Easiest way to test this putting `completion(OnThisDayData().placeholderEntry)` in for `completion` of the following block, then looking at it in the "add widget" system menu (from springboard):
```
func getSnapshot(in context: Context, completion: @escaping (OnThisDayEntry) -> Void) {
        dataStore.fetchLatestAvailableOnThisDayEntry(usingCache: context.isPreview) { entry in
            completion(entry)
        }
    }
```

### CODE REVIEW

**Review by joewalsh — APPROVED** [review]
Looking great! I made some adjustments in eb7297b...1d8888f to fix a few issues I ran into. Here's a quick summary of the changes:
- Updated dates and years to be localized using the content language.
Before:
before
After:
after
- If the user had a primary language that doesn't support on this day, but other languages in the feed, there wouldn't be content:
Screen Shot 2020-09-11 at 12 20 16 PM
After the update, whichever the most preferred language that has a visible on this day card will be shown. This is achieved by using the existing feed cache and `newestVisibleGroup(of:)`. It's is a somewhat major change so I'm happy to back it out if we want to keep the existing strategy for the beta.

**Review by joewalsh — APPROVED** [review]
Reviewed d09526e...4ccded1 and tested by switching primary language - strings were all translated into the app language for French and verified the unsupported error message showed for an unsupported language (Old English). Thanks @mcleinman!

**Review by tonisevener — APPROVED** [review]
Thanks for the thorough test steps! This gives us a good baseline to test all of the widgets off of. This feedback can be handled in a followup PR, I think at this point it's important to get the main feature in so I'm going to approve and merge this.

One additional bit of screenshot feedback is that I noticed the large widget has some cutoff at the bottom when the dynamic type size is increased:

IMG_0158

File: Widgets/Widgets/OnThisDayView.swift [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+        VStack(spacing: 0) {
+            Text(widgetTitle)
+                .foregroundColor(OnThisDayColors.grayColor(colorScheme))
+                .font(.subheadline)
+                .bold()
+                .multilineTextAlignment(.leading)
+                .frame(maxWidth: .infinity, alignment: .leading)
+            Text(monthDay)
+                .font(.title2)
+                .fontWeight(/*@START_MENU_TOKEN@*/.bold/*@END_MENU_TOKEN@*/)
```
[the remark this answers is not in the document]
Comment (mcleinman) [reply Widgets/Widgets/OnThisDayView.swift:249]: Fun fact: SwiftUI has usable defaults - didn't catch that I hadn't updated this one!

**Pull request merged**